{"BEFORE":"        self._window_size = math.ceil(window_size \/ world_size)\n        self._batch_size = batch_size\n        self._tasks = tasks\n        self._compute_mode = compute_mode\n        self._fused_update_limit = fused_update_limit\n        self._should_validate_update = should_validate_update\n        self._default_weights = {}\n        self._update_buffers = {\n            self.PREDICTIONS: [],\n            self.LABELS: [],\n            self.WEIGHTS: [],\n        }\n        if compute_mode == RecComputeMode.FUSED_TASKS_COMPUTATION:\n            task_per_metric = len(self._tasks)\n            self._tasks_iter = self._fused_tasks_iter\n        else:\n            task_per_metric = 1\n            self._tasks_iter = self._unfused_tasks_iter\n\n        self._metrics_computations: nn.ModuleList = nn.ModuleList(\n            [\n                # This Pyre error seems to be Pyre's bug as it can be inferred by mypy\n                # according to https:\/\/github.com\/python\/mypy\/issues\/3048.\n                # pyre-fixme[45]: Cannot instantiate abstract class `RecMetricCoputation`.\n                self._computation_class(\n                    my_rank,\n                    batch_size,\n                    task_per_metric,\n                    self._window_size,\n                    compute_on_all_ranks,\n                    self._should_validate_update,\n                    process_group,\n                    **{**kwargs, **self._get_task_kwargs(task_config)},\n                )\n                for task_config in (\n                    [self._tasks]\n                    if compute_mode == RecComputeMode.FUSED_TASKS_COMPUTATION\n                    else self._tasks\n                )\n            ]\n        )\n","AFTER":"        self._window_size = math.ceil(window_size \/ world_size)\n        self._batch_size = batch_size\n        self._metrics_computations = nn.ModuleList()\n        self._tasks = tasks\n        self._compute_mode = compute_mode\n        self._fused_update_limit = fused_update_limit\n        self._should_validate_update = should_validate_update\n        self._default_weights = {}\n        self._required_inputs = set()\n        self._update_buffers = {\n            self.PREDICTIONS: [],\n            self.LABELS: [],\n            self.WEIGHTS: [],\n        }\n        if compute_mode == RecComputeMode.FUSED_TASKS_COMPUTATION:\n            task_per_metric = len(self._tasks)\n            self._tasks_iter = self._fused_tasks_iter\n        else:\n            task_per_metric = 1\n            self._tasks_iter = self._unfused_tasks_iter\n\n        for task_config in (\n            [self._tasks]\n            if compute_mode == RecComputeMode.FUSED_TASKS_COMPUTATION\n            else self._tasks\n        ):\n            # This Pyre error seems to be Pyre's bug as it can be inferred by mypy\n            # according to https:\/\/github.com\/python\/mypy\/issues\/3048.\n            # pyre-fixme[45]: Cannot instantiate abstract class `RecMetricCoputation`.\n            metric_computation = self._computation_class(\n                my_rank,\n                batch_size,\n                task_per_metric,\n                self._window_size,\n                compute_on_all_ranks,\n                self._should_validate_update,\n                process_group,\n                **{**kwargs, **self._get_task_kwargs(task_config)},\n            )\n            required_inputs = self._get_task_required_inputs(task_config)\n\n            self._metrics_computations.append(metric_computation)\n            self._required_inputs.update(required_inputs)\n\n    def _get_task_kwargs(\n"}